processing/src/main/java/io/druid/query/groupby/epinephelinae/ByteBufferHashTable.java

```diff
@@ -0,0 +1,380 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.epinephelinae;
+
+import com.google.common.primitives.Ints;
+import io.druid.java.util.common.IAE;
+import io.druid.java.util.common.ISE;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferHashTable
```

Comment:
@gianm is concept of regrowth there only because at initializing time we need to set 0 in first byte of all the buckets and it would be too expensive to do that upfront for buffers ? if yes, then, it'll be interesting to see how Unsafe.setMemory(..) performs over 1 GB buffer and if that is fast enough then regrowth business could possibly be removed.